Retro updated

lisazeyen opened the pull request:
update of the build_retro_cost.py script to include the before neglected losses by ventilation and thermal bridges, as well as heat gains (by solar radiation and internal heat gains) in the calculation.

Standard values and building topologies are taken from the  [Episcope/Tabula Project](https://episcope.eu/fileadmin/tabula/public/calc/tabula-calculator.xlsx)

nworbmot commented:
@lisazeyen This is looking great! Sorry it took so long to review. I've tested it and it all works perfectly. Before I pull it into the master could you please make some very minor adjustments:

- In `config.yaml` the switches `'retro_exogen' : False` and `'retro_endogen' : True` contain duplicated information don't they? They would never both be `True`, right? Only one or the other? You could then replace with only one, e.g. `'retro_endogen' : True`.
- In `config.yaml` section on `retrofitting`, make clear which settings apply for exogeneous and which for endogeneous retrofitting, e.g. `dE` is only for exogeneous, right?
- Document changes to retrofitting module in `doc/release_notes.rst`.
- Update the data documentation in `doc/data.csv`, i.e. remove `average_surface_components.csv`, add `tabula-calculator-calcsetbuilding.csv`.
- Include some documentation in `doc/supply_demand.rst` or its own page on how the retrofitting module works, even if it's just a link to the paper [Mitigating heat demand peaks in buildings in a highly renewable European energy system](https://arxiv.org/abs/2012.01831).

lisazeyen commented:
thanks a lot for your feed-back @nworbmot !

- I have renamed in the `config.yaml ` the previously called `'retro_exogen' ` to `'vary_space_heat_exogenously'`.  This factor decreases the space heat demand. This does not necessarily has to represent building renovation, but could also stand for a reduction in space heat demand due to climate change, population decrease, demolition rate and new buildings... Therefore one could also choose both switches `'vary_space_heat_exogenously'` and `'retro_endogen'` to be set to `True.` That would mean we would have a decrease in space heat demand which we get for free (e.g. because of climate change/new buildings) and we still could opimise to reduce the remaining space heat demand with building renovation
- the former option  `'dE'` is now called  `'vary_space_heat_exogenously_factor'`. It does reduce the space heat demand before losses of district heating are applied.
 I updated the space heat demand reductions with more conservative assumptions from the [EUCALC](http://tool.european-calculator.eu/app/buildings/building-types-area/?levers=1j1ff22ffe1g221mofmh31f4vvpnfvwp11C1Chno44fpf111311f21n22h) scenario: LTS baseline.
 This scenario corresponds to : 

> The combination of lever positions under this scenario reproduces, as far as possible, the main sectoral assumptions and key outputs of the LTS Baseline by 2050 published by the European Commission (European Commission 2018).

[European Commission 2018a European long-term strategic vision for a prosperous, modern, competitive and climate neutral economy.](https://ec.europa.eu/clima/sites/clima/files/docs/pages/com_2018_733_analysis_in_support_en_0.pdf)

[EUCALC](http://tool.european-calculator.eu/app/buildings/building-types-area/?levers=1j1ff22ffe1g221mofmh31f4vvpnfvwp11C1Chno44fpf111311f21n22h) has in total 16 different scenarios, it might be interesting to add them as an option to the sector-coupling model (including the CO2 emission pathways, industry assumptions, heat demand reductions, transport demand assumptions).

- I have updated the documentation and the release notes

Maybe there is a better name for `'vary_space_heat_exogenously_factor'` it is a bit long...
